src/sql/src/parsers/tql_parser.rs

```diff
@@ -20,16 +31,58 @@ use sqlparser::tokenizer::Token;
 use crate::error::{self, Result};
 use crate::parser::ParserContext;
 use crate::statements::statement::Statement;
-use crate::statements::tql::{Tql, TqlAnalyze, TqlEval, TqlExplain};
+use crate::statements::tql::{Tql, TqlAnalyze, TqlEval, TqlExplain, TqlParameters};
 
 pub const TQL: &str = "TQL";
 const EVAL: &str = "EVAL";
 const EVALUATE: &str = "EVALUATE";
-const EXPLAIN: &str = "EXPLAIN";
 const VERBOSE: &str = "VERBOSE";
 
+use datatypes::arrow::datatypes::DataType;
 use sqlparser::parser::Parser;
 
+use crate::dialect::GreptimeDbDialect;
+
+pub enum TQLError {
+    Parser(String),
+    Simplification(String),
+    Evaluation(String),
+}
```

Comment:
All errors in this project should be instrumented by `snafu`, you can take an `error.rs` file as a reference. And BTW, it would be good if we put those error-related codes into a dedicated `error.rs` file under the same dir with `tql_parser.rs`.